src/Microsoft.AspNetCore.SystemWebAdapters/Authentication/ResultProcessors/RedirectUrlProcessor.cs

```diff
@@ -7,41 +7,38 @@
 using System.Net.Http;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
-using Microsoft.AspNetCore.WebUtilities;
 
 namespace Microsoft.AspNetCore.SystemWebAdapters.Authentication.ResultProcessors;
 
 /// <summary>
 /// Processes remote authentication results to fix-up 'ReturnUrl' parameters
-/// given in redirect responses so that they will return the user to the correct
-/// original URL instead of to the remote authentication endpoint.
+/// given in redirect responses so that they will return the user to the
+/// ASP.NET Core app rather than URLs with the original ASP.NET app's host.
 /// </summary>
 internal class RedirectUrlProcessor : IRemoteAppAuthenticationResultProcessor
 {
     private const string LocationHeaderName = "Location";
-    private const string ReturnUrlQueryStringName = "ReturnUrl";
 
     /// <summary>
     /// Updates response headers in a remote authentication result so that any
-    /// 'Location' headers will include a ReturnUrl query string corresponding to the
-    /// URL of the original request rather than the URL of the remote authentication endpoint.
+    /// 'Location' headers will redirect to the correct (ASP.NET Core) host.
     /// </summary>
     /// <param name="result">The remote app authentication result to be updated so that redirects point to the right locations.</param>
     /// <param name="context">The HTTP context containing the original request that prompted authentication.</param>
     public Task ProcessAsync(RemoteAppAuthenticationResult result, HttpContext context)
     {
         if (result.ResponseHeaders.TryGetValue(LocationHeaderName, out var locationHeaders))
         {
-            // Look for any Location headers with URLs including a ReturnUrl query string as their value
+            // Look for any Location headers
             var processedRedirectLocations = new List<string>();
```

Comment:
should we defer creating the list until we know it's needed? How often will there be updated locations, but no values?

Reply:
It should almost always be a single string. I'm looping through all values here since headers, generally, can have multiple values but I can't think of any case where it would make sense to have more than one Location header.